artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/JournalStorageManager.java

```diff
@@ -86,6 +87,8 @@
 
    protected ReplicationManager replicator;
 
+   protected final AtomicInteger deletingLargeMessageTasks = new AtomicInteger(0);
```

Comment:
I would put this logic just into `JDBCJournalStorageManager` while exposing here these callbacks:
```java
protected void beforeSubmitDeleteLargeMessageRequest() {
}

protected void afterDeleteLargeMessageCompleted() {
}

protected void afterDeleteLargeMessageError(int code, String errorMessage) {
}
```
The names could be improved, but the idea would be to let just JDBC part to care about this (unless we know that the storage manager need to handle this too), wdyt?

Reply:
TBH, I can't reproduce the issue on JournalStorageManager/JournalImpl but in theory they should have the same issue of JDBCJournalStorageManager/JDBCJournalImpl. I think the issue on JournalStorageManager/JournalImpl is hidden by better performances.

Reply:
@brusdev You can try with mockito to inject slow large messages deletes and see what happen...good point!

Reply:
Take a look on ReusableLatch...

I created ReusableLatch for situations like this.

you make it a final... and you away(0) until it gets to 0.

it's the same overhead as an AtomicINteger
same usability as a CountDownLatch.

Reply:
I replaced AtomicInteger with ReusableLatch.